api/v1alpha1/etcdpeer_types.go

```diff
@@ -11,6 +11,8 @@ import (
 type EtcdPeerSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	BootstrapPeers map[string]string `json:"bootstrapPeers"`
```

Comment:
I think structuring this as:

```yaml
spec:
  bootstrap:
    static:
      peers:
      - ...
```

gives us more flexibility in future in case we need to add additional types of bootstrap, or fields on the current static bootstrap configuration.

Also, do not use `map[string]string` in Kubernetes API types. They require special handling and cause issues in some languages due to their varying unordered nature.

Instead, use a slice of structs with fields. This is also easier to read and gives clearer meaning to users when reading types, as they can clearly see something like:

```yaml
peers:
- name: something
  url: https://...
```

Reply:
Okay. I'll remove this from this RP anyway, and introduce it again when I actually use it. I wasn't aware of issues with `map[string]string`, so I'll implement as you suggest when I do.

Reply:
Removed (for now) in 5e3d260